firecrest/AsyncClient.py

```diff
@@ -826,14 +827,24 @@ async def copy(self, machine: str, source_path: str, target_path: str) -> str:
         self._json_response([resp], 201)
         return target_path
 
-    async def compress(self, machine: str, source_path: str, target_path: str) -> str:
+    async def compress(
+            self,
+            machine: str,
+            source_path: str,
+            target_path: str,
+            fail_on_timeout: bool = True
+    ) -> str:
         """Compress files using gzip compression.
         You can name the output file as you like, but typically these files have a .tar.gz extension.
         When successful, the method returns a string with the path of the newly created file.
 
         :param machine: the machine name where the filesystem belongs to
         :param source_path: the absolute source path
         :param target_path: the absolute target path
+        :param fail_on_timeout: if `True`, the method will raise an exception
+        if the compression fails due to a timeout on the server side.
+        Otherwise, it will submit a job to compress the file and wait for the
```

Comment:
This param description is a bit hard to understand. What about:
if `True` on timeout, this method will raise an exception and won't fall back to submitting a long running job.